sktree/tests/test_honest_forest.py

```diff
@@ -252,3 +255,36 @@ def test_importances(dtype, criterion):
         est.fit(X, y, sample_weight=scale * sample_weight)
         importances_bis = est.feature_importances_
         assert np.abs(importances - importances_bis).mean() < tolerance
+
+
+def test_honest_forest_with_sklearn_trees():
+    """Test against regression in power-curves discussed in:
+    https://github.com/neurodata/scikit-tree/pull/157."""
+
+    # generate the high-dimensional quadratic data
+    X, y = quadratic(1024, 4096, noise=True, seed=0)
+    y = y.squeeze()
+    print(X.shape, y.shape)
+    print(np.sum(y) / len(y))
+
+    clf = HonestForestClassifier(
+        n_estimators=10, tree_estimator=skDecisionTreeClassifier(), random_state=0
+    )
+    honestsk_scores = cross_val_score(clf, X, y, cv=5)
+    print(honestsk_scores)
+
+    clf = HonestForestClassifier(
+        n_estimators=10, tree_estimator=DecisionTreeClassifier(), random_state=0
+    )
+    honest_scores = cross_val_score(clf, X, y, cv=5)
+    print(honest_scores)
+
+    # XXX: surprisingly, when we use the default which uses the fork DecisionTree,
+    # we get different results
+    # clf = HonestForestClassifier(n_estimators=10, random_state=0)
+    # honest_scores = cross_val_score(clf, X, y, cv=5)
+    # print(honest_scores)
+
+    print(honestsk_scores, honest_scores)
+    print(np.mean(honestsk_scores), np.mean(honest_scores))
+    assert_allclose(np.mean(honestsk_scores), np.mean(honest_scores))
```

Comment:
@sampan501 it seems there is some difference in the randomness when we pass in an explicit class vs letting the default handle itself in `HonestForestClassifier`. When you ran quadratic exps, which one did you do?

Reply:
I explicitly passed in `DecisionTreeClassifier()` like you have written above

Reply:
cc: @PSSF23 

In this branch, I basically test:

- sklearn tree vs sklearn-fork tree
- honest tree wrapping sklearn tree vs sklearn-fork tree
- honest forest wrapping sklearn tree vs sklearn-fork tree

In this specific file is the honest forest. We basically get exactly the same performance up to machine precision in trees and almost identical performance in honest forest, so it's confusing me. I think we need a small-scale version of Sambit's power simulation to unit-test and reproduce the performance regression because I'm not seeing any.

Reply:
@sampan501 how many repetitions were you using to compare power curves of the quadratic w/ sklearn-tree vs sklearn-fork tree?

Reply:
I was using about 100 for both. I'm getting consistently higher power using the old method compared to using the `scikit-tree` methods

Reply:
This is now re-filed as an issue in #171 and some additional tests were added in #164